plaidml2/op/lib/ops.cc

```diff
@@ -2469,21 +2469,37 @@ Value squeeze(const Value& value) {
     throw std::runtime_error("Squeeze expects 2 arguments");
   }
 
-  // Read arguments
+  // argument 0: tensor to be squeezed
   auto I = args[0].as_tensor();
-  auto raw_axis = args[1].as_int();
-
-  // Validate arguments
   auto ndims = I.shape().ndims();
-  auto axis = normalize_axis(raw_axis, ndims, "squeeze");
   std::vector<TensorDim> I_dims(ndims);
   std::vector<TensorIndex> I_idxs;
   std::vector<TensorDim> O_dims;
   std::vector<TensorIndex> O_idxs;
   I.bind_dims(I_dims);
+
+  // argument 1: axes to squeeze upon
+  std::vector<int64_t> raw_axes;
+  if (args[1].is_int()) {
+    raw_axes.push_back(args[1].as_int());
+  } else if (args[1].is_none()) {
+    for (uint64_t i = 0; i < ndims; ++i) {
+      // this doesn't work for symbolic dimensions that are later binded to 1
+      if (I_dims[i].as_ptr() && I_dims[i].as_int() == 1) {
```

Comment:
You should use `is_int()` before trying to do `as_int()`, if it's not an int, it will throw during `as_int()`. Also, shouldn't ever be a need to use `as_ptr()`, this is an internal method to get the raw pointer (and should only be used by ffi).

Reply:
Also, probably nicer to say `are later bound` rather than `binded`

Reply:
thanks @flaub that was my bad advice, I'd forgotten `is_int` was a thing